test/apps/iodemo/io_demo.cc

```diff
@@ -1007,25 +1073,76 @@ class DemoServer : public P2pDemoCommon {
     }
 
 private:
-    void save_prev_state() {
-        _prev_state = _curr_state;
+    template<io_op_t op_type> static bool
+    get_min_max(const conn_stat_map_t::iterator& i,
+                conn_stat_map_t::iterator& min,
+                conn_stat_map_t::iterator& max)
+    {
+        long i_bytes  = i->second.get_bytes<op_type>();
+
+        if (i_bytes <= min->second.get_bytes<op_type>()) {
+            min = i;
+            return true;
+        } else if (i_bytes >= max->second.get_bytes<op_type>()) {
+            max = i;
+            return true;
+        } else {
+            return false;
+        }
     }
 
-    void report_state() {
-        LOG << "read:" << _curr_state.read_count -
-                          _prev_state.read_count << " ops, "
-            << "write:" << _curr_state.write_count -
-                           _prev_state.write_count << " ops, "
-            << "connections:" << _curr_state.active_conns << "/"
+    void report_state(double time_interval) {
+        long read_bytes                        = 0;
+        long read_count                        = 0;
+        long write_bytes                       = 0;
+        long write_count                       = 0;
+        conn_stat_map_t::iterator it_read_min  = _conn_stat_map.begin(),
+                                  it_read_max  = _conn_stat_map.begin(),
+                                  it_write_min = _conn_stat_map.begin(),
+                                  it_write_max = _conn_stat_map.begin();
+        for (conn_stat_map_t::iterator it = _conn_stat_map.begin();
+             it != _conn_stat_map.end(); ++it) {
+            bool printable = false;
+
+            read_bytes += it->second.get_bytes<IO_READ>();
+            read_count += it->second.get_completion_count<IO_READ>();
+            printable  |= get_min_max<IO_READ>(it, it_read_min, it_read_max);
+
+            write_bytes += it->second.get_bytes<IO_WRITE>();
+            write_count += it->second.get_completion_count<IO_WRITE>();
+            printable   |= get_min_max<IO_WRITE>(it, it_write_min, it_write_max);
+
+            if (!printable) {
+                it->second.reset();
+            }
```

Comment:
won't it be simpler to reset all stats after we print?
also seems it doesn work if min/max are updated several times (some cntrs are not reset)

Reply:
it would require second loop thru all connections...

Reply:
IIUC, see https://github.com/openucx/ucx/pull/6788/files#r649941166

Reply:
WDYT about possibility that some entries could be not reset?

Reply:
then we can see performance anomaly at least :) for example 2x, 3x, 4X... bandwidth of wire speed

Reply:
it will not be obvious. let's reset stats after each print

Reply:
what do you mean? add another loop after print or keep it as is?

Reply:
add another loop